lib/src/proof.rs

```diff
@@ -0,0 +1,64 @@
+use serde::{Deserialize, Serialize};
```

Comment:
is proofType.rs better?

Reply:
agree, proof_type.rs is better. Updated at 5e4d090d630e35a9a9496b24b4bb24154e2b644f PTAL